packages/plugin-essentials/sources/commands/workspaces/list.ts

```diff
@@ -28,12 +38,24 @@ export default class WorkspacesListCommand extends BaseCommand {
     const configuration = await Configuration.find(this.context.cwd, this.context.plugins);
     const {project} = await Project.find(configuration, this.context.cwd);
 
+    if (configuration.projectCwd === null)
+      throw new UsageError(`This command can only be run from within a Yarn project`);
+
+    const root = this.since ? await gitUtils.fetchRoot(configuration.projectCwd) : null;
+    const base = this.since && root !== null
```

Comment:
After this was moved to `fetchChangedWorkspaces` in 510516aeafc0dd8438cf1bfa2d55aed63415502f I believe this has been cleaned up.
Some of the `this.since` checks were able to be removed as it is already known to be truthy by virtue of calling the function where the type defs require a truthy value.